### ISSUE TITLE
aliasing.unused only removes part of the alias

### ISSUE THREAD

**Issue by deanmorin** [sqlfluff/sqlfluff#4549]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I ran `sqlfluff fix test.sql` and expected the table alias to be removed.

**Note there are 3 empty lines at the top of the file that don't render correctly in markdown.** These are required to reproduce the issue.

test.sql
```sql



select
  foo
from
  bar baz
```

```
L:   6 | P:   5 | AL01 | Implicit/explicit aliasing of table.
                       | [aliasing.table]
L:   6 | P:   5 | AL05 | Alias 'baz' is never used in SELECT statement.
                       | [aliasing.unused]
```

### Expected Behaviour

```
select
  foo
from
  bar
```

### Observed Behaviour

```
select
  foo
from
  bar
az
```

### How to reproduce

This happens every time I run against the initial file. I tried further simplifying my `.sqlfluff` but had trouble figuring out the exact combination required. 

If I add an extra empty lines to the top of the test file I get:

```
select
  foo
from
  bar
baz
```

If I remove a line instead, I get:

```
select
  foo
from
  bar
z
```

### Dialect

snowflake

### Version

```
sqlfluff, version 2.0.0
sqlfluff-templater-dbt, version 2.0.0
Python 3.9.15
dbt 1.4.5
```

### Configuration

.sqlfluff
```
[sqlfluff]
dialect = snowflake
templater = dbt
rules = AL01,AL03,AL04,AL05,AM01,AM07,CP01,CP02,CP03,CP04,CP05,CV01,CV02,CV04,CV05,CV07,CV08,CV11,JJ01,LT01,LT02,LT04,LT06,LT07,LT08,LT10,RF01,RF02,RF03,ST01,ST02,ST03,ST04,ST05,ST07,ST08
large_file_skip_byte_limit = 100000
max_line_length = 100

[sqlfluff:layout:type:comma]
line_position = leading

[sqlfluff:indentation]
indent_unit = space
tab_space_size = 2
indented_joins = true

[sqlfluff:templater:jinja]
apply_dbt_builtins = true

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = lower

[sqlfluff:rules:convention.casting_style]
preferred_type_casting_style = shorthand

[sqlfluff:rules:references.consistent]
single_table_references = unqualified
```

.sqlfluffignore
```
target/
dbt_packages/
macros/
snapshots/
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
